lib/shared/bucketing-assembly-script/__tests__/murmurhash.test.ts

```diff
@@ -1,8 +1,34 @@
-// import { testMurmurhashV3 } from '../build/bucketing-lib.debug'
-// import murmurhash from 'murmurhash'
+import murmurhash from 'murmurhash'
+import { murmurhashV3_js } from '../build/bucketing-lib.debug'
 
-describe('Test testMurmurhashV3', () => {
-    it('should return the correct hash', () => {
-        // expect(testMurmurhashV3('test', 1000)).toEqual(murmurhash.v3('test', 1000))
+function randString(length: number) {
+    let result = ''
+    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()-_=+/.,<>?;:[]{}|\'"'
+    const charactersLength = characters.length
+    for (let i = 0; i < length; i++) {
+        result += characters.charAt(Math.floor(Math.random() * charactersLength))
+    }
+    return result
+}
+
+describe('murmurhash V3 Assembly Script implementation', () => {
+    it('should return the correct hash for simple strings with a varying seed', () => {
+        for (let i = 0; i < 2000; i++) {
+            expect(murmurhashV3_js('some-long-ascii-string', i)).toContain(`${murmurhash.v3('some-long-ascii-string', i)}`)
+            expect(murmurhashV3_js('some-long-ascii-string?', i)).toContain(`${murmurhash.v3('some-long-ascii-string?', i)}`)
+            expect(murmurhashV3_js('some-long-ascii-string!', i)).toContain(`${murmurhash.v3('some-long-ascii-string!', i)}`)
+            expect(murmurhashV3_js('some-long-ascii-string*', i)).toContain(`${murmurhash.v3('some-long-ascii-string*', i)}`)
+        }
+    })
+
+    it('should return the correct hash for a random string with a varying seed', () => {
+        for (let i = 0; i < 20000; i++) {
+            const testString = randString(100)
+            expect(murmurhashV3_js(testString, i)).toContain(`${murmurhash.v3(testString, i)}`)
+        }
+    })
+
+    it('should fail for a non-ascii key', () => {
+        expect(() => murmurhashV3_js('\u11a7', 1)).toThrow('Unsupported character in murmurhashV3 key')
```

Comment:
do we need to add validation to the user id at the DVCUser level?

Reply:
Yeah - this is why I was asking in the standup - we need to force this to be ASCII only.

Reply:
I'll create another ticket for that.